redfish-core/lib/fabric_adapters.hpp

```diff
@@ -236,6 +238,45 @@ inline void doAdapterGet(const std::shared_ptr<bmcweb::AsyncResp>& aResp,
     }
 }
 
+inline void doAdapterPatch(const crow::Request& req,
+			   const std::shared_ptr<bmcweb::AsyncResp>& aResp,
+			   const std::string& adapterId,
+			   const std::string& fabricAdapterPath)
+{
+    std::optional<bool> locationIndicatorActive;
+        
+    if (!json_util::readJsonPatch(req, aResp->res,
+			     "LocationIndicatorActive",
+			     locationIndicatorActive))
+    {
+      return;
+    }
+    
```

Comment:
Can this be easy simple like 
```
inline void doAdapterPatch(const crow::Request& req,
			   const std::shared_ptr<bmcweb::AsyncResp>& aResp,
			   const std::string& adapterId,
			   const std::string& fabricAdapterPath)
{
    std::optional<bool> locationIndicatorActive;

    if (!json_util::readJsonPatch(req, aResp->res,
			     "LocationIndicatorActive",
			     locationIndicatorActive))
    {
      return;
    }
            if (locationIndicatorActive)
            {
                setLocationIndicatorActive(aResp, fabricAdapterPath,
                                           *locationIndicatorActive);
            }
```

Reply:
Do I need the systemBus->async_method_call to send this to the backend?  Frank's changes originally did this, but this code has been refactored since then. So is that not needed>

Reply:
I verified this simplified version does work!  Thanks

Reply:
The current code is still doing another level of GetSubTree inside doAdapterPatch().
So, this would need to be rewritten as to what @gtmills suggested.
Also, readJsonPatch() is better to be invoked at the outer function to avoid the multiple calls inside doAdapterPatch().

If do so, doAdapterPatch() becomes a very small # of codes. 
Then, doAdapterPatch() part can be merged into handleFabricAdapterPatch().

So, this will become like this.

```

inline void
handleFabricAdapterPatch(App &app, const crow::Request &req,
                         const std::shared_ptr<bmcweb::AsyncResp> &aResp,
                         const std::string &systemName,
                         const std::string &adapterId) {
  if (!redfish::setUpRedfishRoute(app, req, aResp)) {
    return;
  }

  std::optional<bool> locationIndicatorActive;

  if (!json_util::readJsonPatch(req, aResp->res, "LocationIndicatorActive",
                                locationIndicatorActive)) {
    return;
  }

  getValidFabricAdapterPath(
      adapterId, systemName, aResp,
      [req, aResp, adapterId, locationIndicatorActive](
          const std::string &fabricAdapterPath,
          [[maybe_unused]] const std::string &serviceName,
          [[maybe_unused]] const dbus::utility::InterfaceList &interfaces) {
        if (locationIndicatorActive) {
          setLocationIndicatorActive(aResp, fabricAdapterPath,
                                     *locationIndicatorActive);
        }
      });
}
```

Reply:
These changes have been made in latest commit.